src/shared/context/product-filter-state.js

```diff
@@ -0,0 +1,16 @@
+const c1 = 'Wedding Sets';
+const c2 = 'Necklace';
+const c3 = 'Rings';
+const c4 = 'Earrings';
+
+export const productFilterState = {
+  categoryFilter: {
+    [c1]: false,
+    [c2]: false,
+    [c3]: false,
+    [c4]: false,
+  },
```

Comment:
In my opinion this is not scalable. Suppose you add a few more categories later then you would have to update the code here added those categories as key, value pairs here. Rather what you could do is create the categoryFilter as an array and add all the selected categories in this array. And then based on the categories present in this categoryFilter array filter out the products having any of these categories. In this case, all edge cases are covered and it's also highly scalable. 
This is just a suggestion so don't do this as of now, try implementing this later if you get time after completing the whole project.

Reply:
Am planning to initialize all category key from api itlsef(not sure about approach yet).
Thanks for suggestion will rethink on approach.

Reply:
> Am planning to initialize all category key from api itlsef(not sure about approach yet). Thanks for suggestion will rethink on approach.

Fixed this issue. Removed hardcoded categories from state & Used object for categoryFilter state instead of array!